### PR TITLE
solved task

### PR DESCRIPTION
**Pull request by slaybrute**

### CODE REVIEW

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 1]
```diff
@@ -1,19 +1,45 @@
...
 import core.basesyntax.Storage;
+import java.util.ArrayList;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+
+    private static final int MAX_STORAGE_VALUE = 10;
+    private int createdStorages;
+
+    private ArrayList<K> keyArrayList = new ArrayList<>();
+    private ArrayList<V> valueArrayList = new ArrayList<>();
```
Comment (okuzan) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:12]: [This is not allowed](https://mate-academy.github.io/jv-program-common-mistakes/java-core/generics/storage.html#dont-use-any-kind-of-list-set-or-map). Create new implementation

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 2]
```diff
@@ -1,19 +1,45 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.util.ArrayList;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+
```
Comment (okuzan) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:7]: delete empty line

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/impl/StorageImpl.java (hunk 1 above)
Comment (slaybrute) [reply src/main/java/core/basesyntax/impl/StorageImpl.java:12]: Okay, you mean It is better to create array with type Object, instead of ArrayList ?

Comment (slaybrute) [reply src/main/java/core/basesyntax/impl/StorageImpl.java]: I've already altered, but applying ArrayList was too much convenient than Array. I can see, than I have added extra code

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 3]
```diff
@@ -1,19 +1,63 @@
...
+    }
+
+    private int indexOf(Object[] array, Object o) {
+        for (int i = 0; i < createdStorages; i++) {
+            if (Objects.equals(array[i], o)) {
+                return i;
+            }
+        }
+        return -1;
+    }
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:31]: All private methods should be below public methods

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 4]
```diff
@@ -1,19 +1,63 @@
...
+    private V value;
+
+    private boolean contains(Object[] array, Object o) {
+        for (int i = 0; i < createdStorages; i++) {
+            if (Objects.equals(array[i], o)) {
+                return true;
+            }
+        }
+        return false;
+    }
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:22]: do not use such naming(o), how to guess what it stores?
```suggestion
    private boolean contains(Object key) {
        for (int i = 0; i < size; i++) {
            if (Objects.equals(keys[i], key)) {
                return true;
            }
        }
        return false;
    }
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 5]
```diff
@@ -1,19 +1,63 @@
...
+
+    private int indexOf(Object[] array, Object o) {
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:24]: do the same as for "private boolean contains(Object key)"

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 6]
```diff
@@ -1,19 +1,63 @@
...
+import java.util.Objects;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_STORAGE_VALUE = 10;
+    private int createdStorages;
+    private Object[] keyArrayList = new Object[MAX_STORAGE_VALUE];
+    private Object[] valueArrayList = new Object[MAX_STORAGE_VALUE];
+
+    private K key;
+    private V value;
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:13]: useless variables
```suggestion
```

**Review by Ivan95kos — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 7]
```diff
@@ -1,19 +1,56 @@
...
+        }
+        return (V) values[indexOf(keys, key)];
     }
 
     @Override
     public int size() {
+        return size;
+    }
+
+    private boolean contains(Object[] array, Object key) {
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:39]: These two methods are similar, try using one.

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 8]
```diff
@@ -1,19 +1,56 @@
...
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_STORAGE_VALUE = 10;
+    private int size;
+    private Object[] keys = new Object[MAX_STORAGE_VALUE];
+    private Object[] values = new Object[MAX_STORAGE_VALUE];
+
     @Override
     public void put(K key, V value) {
+        if (size < MAX_STORAGE_VALUE) {
+            if (contains(keys, key)) {
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:15]: You can use indexOf checking here. If the index is not found, it returns -1. That way you can remove the contains method.

**Review by kozhukhovsky — CHANGES_REQUESTED** [review]

**Review by kozhukhovsky — APPROVED** [review]

**Pull request closed without merging**